src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs

```diff
@@ -114,16 +114,19 @@ protected override Task DisposeCoreAsync(bool initialized)
             }
         }
 
-        private async Task ProjectPropertyChangedAsync(Tuple<ImmutableList<IProjectValueVersions>, TIdentityDictionary> sources)
+        private Task ProjectPropertyChangedAsync(Tuple<ImmutableList<IProjectValueVersions>, TIdentityDictionary> sources)
         {
             IVsProjectRestoreInfo projectRestoreInfo = ProjectRestoreInfoBuilder.Build(sources.Item1);
 
             if (projectRestoreInfo != null)
             {
-                await _solutionRestoreService
-                    .NominateProjectAsync(_projectVsServices.Project.FullPath, projectRestoreInfo, CancellationToken.None)
-                    .ConfigureAwait(false);
+                _projectVsServices.Project.Services.ProjectAsynchronousTasks
+                    .RegisterCriticalAsyncTask(JoinableFactory.RunAsync(() => _solutionRestoreService
```

Comment:
> so multiple duplicate calls would return together

That might be true from Nuget perspective, but CPS will still track all these critical tasks and wait for all them. It might not be an actual issue as all these tasks will complete simultaneously.